src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health) {
+    this.health = health || 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && prey.hidden === false) {
+      prey.health -= 50;
+    }
+
+    if (prey.health <= 0) {
+      const index = Animal.alive.indexOf(prey);
+
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
There is a potential issue when removing the `prey` from the `Animal.alive` array. You should check if `index` is not `-1` before calling `splice` to avoid removing the wrong element or causing an error if `prey` is not found in the array.